decks/card_definitions.json

```diff
@@ -8087,6 +8087,105 @@
+        "colors": ["white"],
+        "rarity": "osr",
+        "life": 5,
+        "actions": [
+            {
+                "skill_id": "mascotcreation",
+                "cost": 2,
+                "limit": "once_per_turn",
+                "effects": [
+                    {
+                        "effect_type": "oshi_activation",
+                        "limit": "once_per_turn",
+                        "skill_id": "mascotcreation",
+                        "pre_effects": [
+                            { "effect_type": "spend_holopower", "amount": 2 },
+                            { "effect_type": "record_used_once_per_turn_effect", "effect_id": "mascotcreation" }
+                        ]
+                    },
+                    {
+                        "effect_type": "choose_cards",
+                        "from": "deck",
+                        "look_at": -1,
+                        "destination": "hand",
+                        "amount_min": 1,
+                        "amount_max": 1,
+                        "requirement": "support",
+                        "requirement_sub_types": ["mascot"],
+                        "reveal_chosen": true,
+                        "remaining_cards_action": "shuffle"
+                    }
+                ]
+            }
+        ],
+        "effects": [
+            {
+                "full_english_text": "FUBUKI_BP02_001_FULL_TEXT",
+                "oshi_effect": true,
+                "skill_id": "fubukingdom",
+                "cost": 2,
+                "limit": "once_per_game",
+                "conditions": [
+                    { "condition": "not_used_once_per_game_effect", "condition_effect_id": "fubukingdom" },
+                    { "condition": "holopower_at_least", "amount": 2 },
+                    { "condition": "performer_is_color", "condition_colors": ["white"] }
+                ],
+                "timing": "on_kill",
+                "effect_type": "choice",
+                "choice": [
+                    {
+                        "oshi_effect": true,
+                        "limit": "once_per_game",
+                        "skill_id": "fubukingdom",
+                        "cost": 2,
+                        "pre_effects": [
+                            {
+                                "effect_type": "oshi_activation",
+                                "limit": "once_per_turn",
+                                "skill_id": "fubukingdom",
+                                "pre_effects": [
+                                    { "effect_type": "spend_holopower", "amount": 2 },
+                                    { "effect_type": "record_used_once_per_game_effect", "effect_id": "fubukingdom" }
+                                ]
+                            }
+                        ],
+                        "effect_type": "multiple_die_roll",
+                        "amount": "per_two_mascots",
+                        "die_effects": [
```

Comment:
Some unexpected work fell on my lap so I can only work on this now.

Anyawy, yeap, we can use this one instead of trying to mangle the single die roll event to work with multiple rolls.